lib/active_merchant/billing/gateways/pagarme/pagarme_recurring_api.rb

```diff
@@ -0,0 +1,137 @@
+          params[:plan_id] = options[:plan_id]
+        end
+
+        if options[:card_id].present?
+          params[:card_id] = options[:card_id]
+        end
+
+        if options[:card_hash].present?
+          params[:card_hash] = options[:card_hash]
+        end
+
+        if options[:card_number].present?
+          params[:card_number] = options[:card_number]
+        end
+
+        if options[:card_holder_name].present?
+          params[:card_holder_name] = options[:card_holder_name]
+        end
+
+        if options[:card_expiration_date].present? && expiration_date(options[:card_expiration_date])
+          params[:card_expiration_date] = options[:card_expiration_date]
+        end
+
+        commit(:put, "subscriptions/#{invoice_id}", params)
+      end
+
+      def cancel(invoice_id)
+        params = {}
+        commit(:post, "subscriptions/#{invoice_id}/cancel", params)
+      end
+
+
+      def invoice(invoice_id)
+        response = commit(:get, "transactions/#{invoice_id}", nil)
+        Response.new(true, nil, {invoice: invoice_to_response(response.params)})
+      end
+
+
+      def invoices(subscription_id)
+        response = service_pagarme.invoices_by_subscription(subscription_id)
+        Response.new(true, nil, {invoices:  invoices_to_response(response)})
+      end
+
+      def payments(invoice_id)
+        response = service_pagarme.payments_from_invoice(invoice_id)
+        Response.new(true, nil, { payments: payments_to_response(response) })
+      end
+
+      def payment(invoice_id, payment_id)
+        response = service_pagarme.payment_from_invoice(invoice_id, payment_id)
+        Response.new(true, nil, { payment: payment_to_response(response) })
+      end
+
+      def subscription_details(subscription_code)
+        response = PagarMe::Subscription.find_by_id(subscription_code)
+        Response.new(true, nil, subscription_response(response))
+      end
+
+      private
+
+      def expiration_date(date)
+        if /((1[0-2]|0[1-9])([0-9]){2})/ =~ date
+          date
+        else
+          raise "Data de expiração do cartão com formato inválido"
+        end
+      end
+
+      def ensure_customer_created(options)
+        customer_response(PagarMe::Customer.find_by_id(options[:customer][:id]))
+      rescue
+        create_customer(options[:customer], options[:customer][:address])
+      end
+
+      def create_customer(customer, address)
+        params = customer_params(customer, address)
+        PagarMe::Customer.new(params).create.to_hash
+      end
+
+      def ensure_plan_created(plan_code, amount, options)
+        plan_response(PagarMe::Plan.find_by_id(plan_code))
+      rescue
+        create_plan(options, amount)
+      end
+
+      def create_plan(params, amount)
+        requires!(options, :name, :days)
+
+        params = {
+            :name => params[:name],
+            :days => params[:days],
+            :amount => amount,
+        }
+        PagarMe::Plan.new(params).to_hash
```

Comment:
@mandado faltou o update do plano.. E outra coisa, quando você for criar o plano no pagar.me, precisa enviar o código, somos nós que criamos esse código pra poder buscar depois.. 

No caso do moip por exemplo, sempre que vamos criar uma assinatura, é verificado se existe o plano (através do code https://github.com/Edools/active_merchant/blob/master/lib/active_merchant/billing/gateways/moip/moip_recurring_api.rb#L54). Se ele existir, então verificamos se teve alteração, e se tiver, então é atualizado no moip. Se não existir ele é criado com o nosso code.